### DISCUSSION TITLE
Problem with debug mode

### DISCUSSION THREAD

**Discussion by mrdenis73** [roadrunner-server/roadrunner#1890] [General]
The worker does not have time to send a message to Kafka in debug mode

Is it possible to set the time for a worker to be removed or how to solve the problem without crutches?

The request arrives via http, some business logic is done, sent to Kafka and a response is immediately returned.

http:
  address: 0.0.0.0:8080
  pool:
    num_workers: 1
    #debug: true # works specifically, ignores all settings in the pool section and starts workers upon an incoming request, deletes the worker after the response
                 # producer may not have time to write to Kafka

HttpController.

        $this->paymentService->produce($xpayRequest, (string) $requestQueue->getId());  // produce Kafka

        usleep(5001); // works with a delay 

        return new JsonResponse(['queueId' => $orderRequest->getId()]);

**Comment by rustatian** [comment]
Hey @mrdenis73 👋 
In the debug mode, workers are destroyed immediately after returning a response to RR. There are no timeouts.
In this specific case, it is better to use a regular worker w/o debug mode and use `./rr reset` if you need to restart the worker on the code change.

**Reply by mrdenis73** [reply]
Hello! No, you don't need to restart anything. A request is sent via http, then a request is made to Kafka and a response is immediately given. The worker seems to be deleted so quickly that it does not have time to write to Kafka, looks like a bug.

I make a delay, everything is ok, and in prod mode everything works fine too.

**Reply by rustatian** [reply]
I need an example repository with a reproduction of that problem to move further (without frameworks, preferably). Because, you know, worker can't be deleted before it returns a response to RR. Since you have 2 different worker types (http + jobs), the logic should be as following:
1. HTTP request received. Worker allocated.
2. Request sent to the Kafka broker, new worker allocated (jobs worker).
3. HTTP worker, if you're still in it, should not exit (you may check that by sending some log before exit).
4. Jobs worker processes request and gets destroyed. Keep in mind, there are no response to the RPC or other worker, except Kafka broker (ack/nack/retry).
